gravitee-apim-console-webui/src/management/api/api-traffic-v4/analytics/components/api-analytics-proxy-filter-bar/api-analytics-proxy-filter-bar.component.ts

```diff
@@ -242,4 +280,57 @@ export class ApiAnalyticsProxyFilterBarComponent implements OnInit {
       ...partial,
     });
   }
+
+  private getApplicationChips$(applications: string[] | null): Observable<FilterChip[]> {
+    if (!applications?.length) {
+      return of([]);
+    }
+
+    const cachedChips: FilterChip[] = [];
+    const uncachedIds: string[] = [];
+
+    applications.forEach((appId) => {
+      if (this.applicationChipCache[appId]) {
+        cachedChips.push(this.applicationChipCache[appId]);
+      } else {
+        uncachedIds.push(appId);
+      }
+    });
+
+    if (uncachedIds.length === 0) {
+      return of(cachedChips);
+    }
+
+    return this.fetchApplicationsAndUpdateCache$(uncachedIds, cachedChips);
+  }
+
+  private fetchApplicationsAndUpdateCache$(uncachedIds: string[], cachedChips: FilterChip[]): Observable<FilterChip[]> {
+    return this.applicationService.findByIds(uncachedIds, 1, 200).pipe(
+      map((response) => {
+        return uncachedIds.map((id) => {
+          const foundApp = response.data.find((app) => app.id === id);
+          return {
+            key: 'applications',
+            value: id,
+            display: foundApp?.name ?? 'Unknown Application',
+          };
+        });
+      }),
+      catchError(() => {
```

Comment:
As discussed it may be removed or we could simple use the snack bar here as we are doing a findByIds.